### PR TITLE
Add spell correction for CLI options

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
This gives you some suggestions if you’ve mistyped a command line parameter.

E.g. on `cbmc --traec` it will suggest `--trace`, if you type `cbmc --cvc` it will suggest trying `--cvc3` or `--cvc4`, that sort of thing.

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: regression/cbmc/unknown-argument-suggestion/test.desc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CORE
+dummy.c
+--traec
+did you mean --trace
+^EXIT=[123456789]\d*$
```
Comment (hannes-steffenhagen-diffblue) [inline regression/cbmc/unknown-argument-suggestion/test.desc]: A right nuisance: Apparently the error codes on windows and linux are different, but our test.pl script insist on very particular exit code checks. I'm tempted to just remove this test to be honest.

File: src/util/nfa.h [hunk 2]
```diff
@@ -0,0 +1,168 @@
+/// \file
+/// \author Diffblue Ltd.
+///
+/// A simple NFA implementation
```
Comment (martin-cs) [inline src/util/nfa.h]: Please could you add what it is intended for.  There have been other automaton implementations in CPROVER (for SATAbs for example) and I wouldn't like future people to get the wrong idea of what this is.

**Review by martin-cs — APPROVED** [review]
This is a really cute feature -- we should have this.

File: src/util/nfa.h (hunk 2 above)
Comment (hannes-steffenhagen-diffblue) [reply src/util/nfa.h]: Good point, will do

**Review by chrisr-diffblue — APPROVED** [review]
While this is a bit of feature creep - it is indeed cute, but also really helpful to new users (and old users that typo :-) ) so I'm happy for it to go in.

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**